diff --git a/GPClient/normalloginwindow.cpp b/GPClient/normalloginwindow.cpp
index 1626a09..41a153e 100644
--- a/GPClient/normalloginwindow.cpp
+++ b/GPClient/normalloginwindow.cpp
@@ -2,6 +2,9 @@
 
 #include "normalloginwindow.h"
 #include "ui_normalloginwindow.h"
+#include "gphelper.h"
+
+using namespace gpclient::helper;
 
 NormalLoginWindow::NormalLoginWindow(QWidget *parent) :
     QDialog(parent),
@@ -11,6 +14,9 @@ NormalLoginWindow::NormalLoginWindow(QWidget *parent) :
     setWindowTitle("GlobalProtect Login");
     setFixedSize(width(), height());
     setModal(true);
+
+    ui->username->setText(settings::get("username", "").toString());
+    ui->password->setText(settings::get("password", "").toString());
 }
 
 NormalLoginWindow::~NormalLoginWindow()
@@ -50,6 +56,10 @@ void NormalLoginWindow::on_loginButton_clicked()
     const QString username = ui->username->text().trimmed();
     const QString password = ui->password->text().trimmed();
 
+    // save username and password (not secure tho)
+    settings::save("username", username);
+    settings::save("password", password);
+
     if (username.isEmpty() || password.isEmpty()) {
         return;
     }
